test/ruby/test_iseq.rb

```diff
@@ -187,8 +187,8 @@ def test_frozen_string_literal_compile_option
     s1, s2, s3, s4 = compile(code, line, {frozen_string_literal: true}).eval
     assert_predicate(s1, :frozen?)
     assert_predicate(s2, :frozen?)
-    assert_predicate(s3, :frozen?)
-    assert_predicate(s4, :frozen?)
+    assert_not_predicate(s3, :frozen?)
+    assert_predicate(s4, :frozen?) # should probably not be frozen, but unrealistic code
```

Comment:
I couldn't find what optimizes `s = "#{'foo'}"` to just `putobject "foo"`.
Ideally it should not be frozen.
OTOH, I believe nobody write such code (but rather `"#{s}"`).

Reply:
@nobu has a fix for this: https://bugs.ruby-lang.org/issues/17104#note-20
I'm not sure if it's worth the added complexity.